### PR TITLE
[HELIX-592] addCluster should respect overwriteExisitng when adding stateModel Definations.

### PR DESCRIPTION
**Pull request by lei-xia**
Congrui Ji has sent this fix before, and got some comments.  But he is on vacation, and we (LinkedIn) do need the fix ASAP, so I am sending this again.  Thanks

### CODE REVIEW

File: helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java [hunk 1]
```diff
@@ -715,14 +715,26 @@ public ExternalView getResourceExternalView(String clusterName, String resourceN
...
     if (_zkClient.exists(stateModelPath)) {
-      logger.warn("Skip the operation.State Model directory exists:" + stateModelPath);
-      throw new HelixException("State model path " + stateModelPath + " already exists.");
+      if (recreateIfExists) {
+        logger.warn("Operation.State Model directory exists:" + stateModelPath +
+            ", remove and recreate.");
+        _zkClient.deleteRecursive(stateModelPath);
+      } else {
+        logger.warn("Skip the operation.State Model directory exists:" + stateModelPath);
+        return;
```
Comment (kanakb) [inline helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java]: Why was this changed to no longer throw an exception? It's better to fail loudly in methods that return `void`.

File: helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java [hunk 2]
```diff
@@ -715,14 +715,26 @@ public ExternalView getResourceExternalView(String clusterName, String resourceN
...
     if (!ZKUtil.isClusterSetup(clusterName, _zkClient)) {
       throw new HelixException("cluster " + clusterName + " is not setup yet");
     }
     String stateModelDefPath = HelixUtil.getStateModelDefinitionPath(clusterName);
     String stateModelPath = stateModelDefPath + "/" + stateModelDef;
     if (_zkClient.exists(stateModelPath)) {
-      logger.warn("Skip the operation.State Model directory exists:" + stateModelPath);
-      throw new HelixException("State model path " + stateModelPath + " already exists.");
+      if (recreateIfExists) {
+        logger.warn("Operation.State Model directory exists:" + stateModelPath +
```
Comment (kanakb) [inline helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java]: Change to `info` log level.

File: helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java (hunk 1 above)
Comment (lei-xia) [inline helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java]: This is to align with the behavior of addCluster,  which return success if the cluster exists and overwrite flag is false.

File: helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java (hunk 2 above)
Comment (lei-xia) [inline helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java]: Fixed in new diff.  Thanks!

**Comment by kanakb** [comment]
This closes #26.
